lxc fails to build when the Landlock header is missing and `landlock-monitor` is `false`

For example it fails to compile on OSS-Fuzz because of that: https://github.com/lxc/lxc/actions/runs/18479871665/job/52652294373:
```sh
...
clang -Isrc/lxc/liblxc.a.p -Isrc/lxc -I../src/lxc -I. -I.. -Isrc -I../src -Isrc/include -I../src/include -I../src/lxc/cgroups -I../src/lxc/storage -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=gnu11 -O0 -g -Wno-format-signedness -Wno-missing-field-initializers -Wno-unused-parameter -Wvla -Wcast-align -Wstrict-prototypes -fno-strict-aliasing -fstack-clash-protection --param=ssp-buffer-size=4 -Werror=implicit-function-declaration -Wmissing-include-dirs -Wold-style-definition -Winit-self -Wunused-but-set-variable -Wno-unused-parameter -Wfloat-equal -Werror=return-type -Werror=incompatible-pointer-types -Wformat=2 -Wshadow -Wendif-labels -Werror=overflow -fdiagnostics-show-option -Werror=shift-count-overflow -Wdate-time -Wnested-externs -fasynchronous-unwind-tables -fexceptions -Warray-bounds -Wreturn-local-addr -Wno-typedef-redefinition -Wno-gnu-variable-sized-type-not-at-end -ffat-lto-objects -include config.h -O1 -fno-omit-frame-pointer -gline-tables-only -Wno-error=incompatible-function-pointer-types -Wno-error=int-conversion -Wno-error=deprecated-declarations -Wno-error=implicit-function-declaration -Wno-error=implicit-int -Wno-error=unknown-warning-option -Wno-error=vla-cxx-extension -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -fPIC -pthread -fvisibility=default -DIN_LIBLXC -MD -MQ src/lxc/liblxc.a.p/start.c.o -MF src/lxc/liblxc.a.p/start.c.o.d -o src/lxc/liblxc.a.p/start.c.o -c ../src/lxc/start.c
../src/lxc/start.c:9:10: fatal error: 'linux/landlock.h' file not found
    9 | #include <linux/landlock.h>
      |          ^~~~~~~~~~~~~~~~~~
1 error generated.
```
In a couple of weeks It should be possible to bump the base builder images from Ubuntu 20.04 to Ubuntu 24.04 (where `linux-libc-dev` is new enough and comes with the header)) but it seems it shouldn't fail to compile when `landlock-monitor` is `false` regardless of whether the header is present or not.

--- a/src/lxc/start.c
+++ b/src/lxc/start.c
@@ -6,7 +6,6 @@
 #include <errno.h>
 #include <fcntl.h>
 #include <grp.h>
-#include <linux/landlock.h>
 #include <poll.h>
 #include <pthread.h>
 #include <signal.h>
@@ -69,6 +68,8 @@
 #endif
 
 #if HAVE_LANDLOCK_MONITOR
+#include <linux/landlock.h>
+
 #ifndef landlock_create_ruleset
 static inline int
 landlock_create_ruleset(const struct landlock_ruleset_attr *const attr,

